src/neon/deser.rs

```diff
@@ -0,0 +1,204 @@
+                ///////////////////////
+                // Above, check for overflow in case someone has a crazy string (>=4GB?)
+                // But only add the overflow check when the document itself exceeds 4GB
+                // Currently unneeded because we refuse to parse docs larger or equal to 4GB.
+                ////////////////////////
+
+                // we advance the point, accounting for the fact that we have a NULl termination
+
+                len += quote_dist as usize;
+                unsafe {
+                    let v = self.input.get_unchecked(idx..idx + len) as *const [u8] as *const str;
+                    return Ok(&*v);
+                }
+
+                // we compare the pointers since we care if they are 'at the same spot'
+                // not if they are the same value
+            }
+            if (quote_bits.wrapping_sub(1) & bs_bits) != 0 {
+                // Move to the 'bad' character
+                let bs_dist: u32 = trailingzeroes(u64::from(bs_bits));
+                len += bs_dist as usize;
+                src_i += bs_dist as usize;
+                break;
+            } else {
+                // they are the same. Since they can't co-occur, it means we encountered
+                // neither.
+                src_i += 16;
+                len += 16;
+            }
+        }
+
+        let mut dst_i: usize = 0;
+        let dst: &mut [u8] = &mut self.strings;
+
+        loop {
+            let v: int8x16_t = if src.len() >= src_i + 16 {
+                // This is safe since we ensure src is at least 16 wide
+                #[allow(clippy::cast_ptr_alignment)]
+                    unsafe {
+                    *(src.as_ptr().add(src_i) as *const int8x16_t)
+                }
+            } else {
+                unsafe {
+                    padding
+                        .get_unchecked_mut(..src.len() - src_i)
+                        .clone_from_slice(src.get_unchecked(src_i..));
+                    // This is safe since we ensure src is at least 16 wide
+                    #[allow(clippy::cast_ptr_alignment)]
+                        *(padding.as_ptr() as *const int8x16_t)
+                }
+            };
+
+            #[allow(clippy::cast_ptr_alignment)]
+                unsafe {
+                *(dst.as_mut_ptr().add(dst_i) as *mut int8x16_t) = v;
+            };
+
+            // store to dest unconditionally - we can overwrite the bits we don't like
+            // later
+            let bs_bits: u32 = 0; // FIXME
+//                unsafe {
+//                static_cast_u32!(_mm_movemask_epi8(vceqq_s8(
+//                    v,
+//                    vdupq_n_s8(b'\\' as i8)
+//                )))
+//            };
+            let quote_mask = 0; // FIXME: unsafe { vceqq_s8(v, vdupq_n_s8(b'"' as i8)) };
+            let quote_bits = 0; // FIXME: unsafe { static_cast_u32!(_mm_movemask_epi8(quote_mask)) };
+            if (bs_bits.wrapping_sub(1) & quote_bits) != 0 {
```

Comment:
Hm... @Licenser would you happen to know the ARM idiom for this movemask sequence?

Reply:
Looking at the upstream code https://github.com/lemire/simdjson/blob/d9a0e2b8f441c20ad46276fdb8ce24f2aebdc07b/include/simdjson/stringparsing_arm64.h#L40 those two lines seem to be about setting the quote bits and mask